File: HorseRacing.Contracts/Models/Game/Responses/GetLobbyUsersWithBets/GetLobbyUsersWithBetsResponseDto.cs
using HorseRacing.Contracts.Base.Dto;
using HorseRacing.Contracts.Models.Game.Dtos;

namespace HorseRacing.Contracts.Models.Game.Responses.GetLobbyUsersWithBets
{
    public class GetLobbyUsersWithBetsResponseDto : BaseDto
    {
        /// <summary>
        /// Код игры
        /// </summary>
        public Guid GameId { get; set; }
        /// <summary>
        /// Наименование игры
        /// </summary>
        public string GameName { get; set; } = string.Empty;
        /// <summary>
        /// Общий банк ставок
        /// </summary>
        public decimal TotalBank { get; set; }
        /// <summary>
        /// Список пользователей в лобби с их ставками
        /// </summary>
        public List<GameUserDto> Players { get; set; } = new();
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/responses/get-lobby-users-with-bets/get-lobby-users-with-bets-response-dto.ts
import type { BaseDto } from '~/interfaces/api/contracts/base/dto/base-dto';
import type { GameUserDto } from '../../dto/game-user-dto';

export interface GetLobbyUsersWithBetsResponseDto extends BaseDto {
    /** Код игры */
    GameId: string;
    /** Наименование игры */
    GameName: string;
    /** Общий банк */
    TotalBank: number;
    /** Список игроков */
    Players: Array<GameUserDto> | [];
}
